File: leetcode/705. Design HashSet/hashSet.ts
class MyHashSet {
  dictionary: Record <number, number>
  length: number

  constructor() {
    this.dictionary = {}
    this.length = 0
  }

  add(key: number): void {
    if (!this.contains(key)) {
      this.dictionary[key] = key
      this.length++
    }
  }

  remove(key: number): void {
    if (this.contains(key)) {
      delete this.dictionary[key]
      this.length--
    }
  }

  contains(key: number): boolean {
      return this.dictionary[key] !== undefined
  }

  values(): number[] {
    return Object.values(this.dictionary)
  } 
}

File: LeetCode/705. Design HashSet/hashSet.cs
public class MyHashSet {
    private int length;
    private Dictionary<int, int> hash;
    
    public MyHashSet() {
        length = 0;
        hash = new Dictionary<int, int>();
    }
    
    public void Add(int key) {
        if (!hash.ContainsKey(key))
        {
            hash[key] = key;
            length++;
        }
    }
    
    public void Remove(int key) {
        if (hash.ContainsKey(key))
        {
            hash.Remove(key);
            length--;
        }
    }
    
    public bool Contains(int key) {
        return hash.ContainsKey(key);
    }

  public int Size => length;
}
